targets/arm/mikroe/sam/src/gpio/implementations/implementation_1/hal_ll_gpio_port.c

```diff
@@ -0,0 +1,299 @@
+}
+
+void hal_ll_gpio_module_struct_init( module_struct const *module, bool state )
+{
+    int8_t index = 0;
+
+    while ( module->pins[ index ] != GPIO_MODULE_STRUCT_END )
+    {
+        hal_ll_gpio_config_pin_alternate_enable( module->pins[ index ], module->configs[ index ], state );
+
+        index++;
+    }
+}
+
+/**
+  * @brief  Enable gpio port clock
+  * @param  port - port base address
+  * @return none
+  */
+static void hal_ll_gpio_clock_enable( hal_ll_gpio_base_t *port )
+{
+    uint32_t pos = 0;
+
+    switch ( ( uint32_t )port )
+    {
+        #ifdef __PORT_A_CN
+        case HAL_LL_GPIOA_BASE_ADDR: pos = HAL_LL_PMC_ID_PIOA_BIT;   break;
+        #endif
+        #ifdef __PORT_B_CN
+        case HAL_LL_GPIOB_BASE_ADDR: pos = HAL_LL_PMC_ID_PIOB_BIT;   break;
+        #endif
+        #ifdef __PORT_C_CN
+        case HAL_LL_GPIOC_BASE_ADDR: pos = HAL_LL_PMC_ID_PIOC_BIT;   break;
+        #endif
+        #ifdef __PORT_D_CN
+        case HAL_LL_GPIOD_BASE_ADDR: pos = HAL_LL_PMC_ID_PIOD_BIT;   break;
+        #endif
+        #ifdef __PORT_E_CN
+        case HAL_LL_GPIOE_BASE_ADDR: pos = HAL_LL_PMC_ID_PIOE_BIT;   break;
+        #endif
+    }
+
+    set_reg_bit( _PMC_PCER0, pos );
+}
+
+static void hal_ll_gpio_config( hal_ll_gpio_base_t *port, hal_ll_gpio_mask_t pin_mask, uint32_t config )
+{
+    uint8_t  pin_num     = 0;
+    uint32_t pin_pos     = 0;
+    uint32_t current_pin = 0;
+    uint32_t port_base   = 0;
+
+    hal_ll_gpio_base_handle_t *port_ptr;
+
+    port_ptr = ( hal_ll_gpio_base_handle_t *)port;
+
+    hal_ll_gpio_clock_enable( port );
+
+    // Check if it is a digital signal or alternate function
+    if ( ( config & GPIO_CFG_MODE_ANALOG ) != GPIO_CFG_MODE_ANALOG ) {
+        if ( config & GPIO_CFG_MODE_INPUT ) {
+            // Set gpio direction as input with filtering.
+            port_ptr->pio.enable |= pin_mask;
+            port_ptr->filter.enable |= pin_mask;
+        } else {
+            // Set gpio direction as output and enable Pull Up resistor.
+            port_ptr->pio.enable |= pin_mask;
+            port_ptr->output.enable |= pin_mask;
+            port_ptr->pullup_off.disable |= pin_mask;
+        }
+    }
+    else {
+        // Set pin to analog.
+        port_ptr->pio.disable = pin_mask;
+    }
+}
+
+/**
+  * @brief  Configure port pins alternate
+  *         functions
+  * @param  module_pin - desired pin
+  *         config     - pin settings
+  * @return none
+  */
+static void hal_ll_gpio_config_pin_alternate_enable( uint8_t module_pin, uint8_t module_config, bool state )
+{
+    hal_ll_pin_name_t pin_index;
+    hal_ll_gpio_base_handle_t *port_ptr;
+
+    // Get the port address.
+    port_ptr = ( hal_ll_gpio_base_handle_t *)hal_ll_gpio_port_base( hal_ll_gpio_port_index ( module_pin ) );
+
+    // Get the pin index.
+    pin_index = hal_ll_gpio_pin_index( module_pin );
+
+    // Disable PIO control over this pin.
+    port_ptr->pio.disable |= 1 << pin_index;
+
+    switch ( module_config ) {
```

Comment:
Again, add a default case.

Reply:
Added it for all source files where needed.